src/utils/fundamentals/fundamentals.js

```diff
@@ -0,0 +1,199 @@
+   }else{
+    code=codes
+   }
+   
+   let invalidDate=""
+   if(invalidDates){
+    if(Array.isArray(invalidDates)){
+      for(var i = 0; i < invalidDates.length;i++) {
+            invalidDate+=invalidDates[i]+","
+       } 
+     invalidDate = invalidDate.substring(0,invalidDate.length - 1);
+    }else{
+      invalidDate=invalidDates
+    }
+   }
+    
+   let params 
+   if(!invalidDates){
+          params = {
+          AMaaSClass: 'fundamentalBusinessDate',
+          query: {startDate: [date], countryCode: [code], offset: [offset]}
+          }
+   }else{
+         params = {
+          AMaaSClass: 'fundamentalBusinessDate',
+          query: {startDate: [date], countryCode: [code], offset: [offset], invalidDate: [invalidDate]}
+          }
+   }
+
+    let promise = searchData(params).then(result => {
+    if (typeof callback === 'function') {
+      callback(null, result)
+    } 
+      return result 
+    }).catch((err) => {
+      console.log(err.message);
+    });
+   if (typeof callback !== 'function') {
+    // return promise if callback is not provided
+    return promise
+   }
+  promise.catch(error => callback(error))
+}
+
+/**
+ * Make request and search data
+ * @function processDateInfo
+ * @param {object} params - object of parameters:
+ * @param {string} params.codes - A country code string or an array of country codes strings 
+ * @param {string} params.date - start date
+ * @param {function} [callback] - Called with two arguments (error, result) on completion. `result` is a country name returned or country names. Omit to return Promise
+ * @returns {Promise|null} 
+ */
+export function processDateInfo ({date, codes}, callback)
+{
+    let code=""
+    if(Array.isArray(codes)){
+      for(var i = 0; i < codes.length;i++) {
+            code+=codes[i]+","
+       } 
+     code = code.substring(0, code.length - 1);
+   }else{
+    code=codes
+   }
+
+    const params = {
+          AMaaSClass: 'fundamentalDateInfo',
+          query: {startDate: [date], countryCode: [code]}
+    }
+   let promise = searchData(params).then(result => {
+    if (typeof callback === 'function') {
+      callback(null, result)
+    } 
+      return result 
+    }).catch((err) => {
+      console.log(err.message);
+    });
+   if (typeof callback !== 'function') {
+    // return promise if callback is not provided
+    return promise
+   }
+  promise.catch(error => callback(error))
+}
+
+/**
+ * Make request and search data
+ * @function holidays
+ * @param {object} params - object of parameters:
+ * @param {string} params.codes - A country code string or an array of country codes strings 
+ * @param {string} params.year - A year string or an array of 
+ * @param {function} [callback] - Called with two arguments (error, result) on completion. `result` is a country name returned or country names. Omit to return Promise
+ * @returns {Promise|null} 
+ */
+export function holidays ({codes, years}, callback)
```

Comment:
You can go through this last function yourself and make the same changes I mentioned above: namely the array checking mechanism and removing the catch clause from the promise.